### ISSUE TITLE
Super resolution using Deep CNN's

### ISSUE THREAD

**Issue by GilLevi** [vlfeat/matconvnet#66]
Hi,

I'm trying to reproduce the method described in the following paper: 
Dong, Chao, et al. "Learning a deep convolutional network for image super-resolution." Computer Vision–ECCV 2014. Springer International Publishing, 2014. 184-199.‏

The method (if I understood it correctly) is to use two conv layers where the second one contains filters of size 1x1xcx1, thus producing images of the same size as the input images.

For some reason, the loss does not seem decrees. I defined the loss functions as follows:

function Y = vl_nnRMSEloss(X,c,dzdy)

if nargin <= 2
    a=(X-c).^2;
    Y=sum(a(:));
    Y=Y/size(X,4);
    
else
    Y = +((X-c))*dzdy;
end

And I'm using the following code to train the net:

data=zeros(140,140,1,M);
% filling in Data ...

imdb.images.data = imresize(imresize(data,0.5),2);
imdb.images.data_mean = mean(data,4);
imdb.images.labels = data ;
imdb.images.set=[ones(1,3*M/5), 2*ones(1,1*M/5) , 3*ones(1,1*M/5)];

imdb.images.data = single(imdb.images.data);
imdb.images.data_mean = single(imdb.images.data_mean);
imdb.images.labels = single(imdb.images.labels);

% Define network architecture
net.layers = {} ;

filters1=zeros(5,5,1,32);
for ix=1:32
    filters1(:,:,1,ix)=fspecial('gaussian',5,randi(20));
end

filters2=zeros(1,1,32,1);
for ix=1:32
    filters2(:,:,32,:)=fspecial('gaussian',1,randi(20));
end

filters1=single(filters1);
filters2=single(filters2);

% 1 conv1
net.layers{end+1} = struct('type', 'conv', ...
  'filters', filters1 , ...
  'biases', zeros(1, 32, 'single'), ...
  'filtersLearningRate', 1, ...
  'biasesLearningRate', 2, ...
  'stride', 1, ...
  'pad', 1) ;

% 2 conv2
net.layers{end+1} = struct('type', 'conv', ...
  'filters', filters2 , ...
  'biases', zeros(1, 1, 'single'), ...
  'filtersLearningRate', 1, ...
  'biasesLearningRate', 2, ...
  'stride', 1, ...
  'pad', 1) ;   

% 4 RELU
net.layers{end+1} = struct('type', 'relu') ;

% 4 RMSE loss
net.layers{end+1} = struct('type', 'RMSEloss') ;

opts.train.batchSize = 100 ;
opts.train.numEpochs = 20 ;
opts.train.continue = false ;
opts.train.useGpu = false ;
opts.train.learningRate = [0.001*ones(1, 12) 0.0001*ones(1,6) 0.00001] ;

[net, info] = cnn_train(net, imdb, @getBatch, ...
    opts.train, ...
    'val', find(imdb.images.set == 2)) ;

Moreover, when calculating the derivative of the first conv layer (backprop):
 switch l.type
            case 'conv'
                [res(i).dzdx, res(i).dzdw{1}, res(i).dzdw{2}] = ...
                    vl_nnconv(res(i).x, l.filters, l.biases, ...
                    res(i+1).dzdx, ...
                    'pad', l.pad, 'stride', l.stride) ;

The values of res(i).dzdw{1} are all zero, except for the last filter.

Can anyone please provide some insight as to what might be the problem?

Thanks!
Gil
